tests/unit/h/schemas/api/group_test.py

```diff
@@ -40,6 +40,32 @@ def test_it_raises_if_name_too_long(self, schema):
         with pytest.raises(ValidationError, match="name:.*is too long"):
             schema.validate({"name": "o" * (GROUP_NAME_MAX_LENGTH + 1)})
 
+    def test_it_raises_if_name_isnt_a_string(self, schema):
+        name = 42
+
+        with pytest.raises(
+            ValidationError, match=f"name: {name} is not of type 'string'"
+        ):
+            schema.validate({"name": name})
+
+    @pytest.mark.parametrize(
+        "name",
+        [
+            " " * GROUP_NAME_MIN_LENGTH,
+            ("o" * (GROUP_NAME_MIN_LENGTH - 1)) + " ",
+        ],
+    )
+    def test_it_raises_if_name_is_too_short_after_trimming_whitespace(
+        self, schema, name
+    ):
+        with pytest.raises(ValidationError, match="name:.*is too short"):
+            schema.validate({"name": name})
+
+    def test_it_validates_with_no_data(self, schema):
+        appstruct = schema.validate({})
+
+        assert appstruct == {}
```

Comment:
`GroupAPISchema` is the base schema that `CreateGroupAPISchema` and `UpdateGroupAPISchema` inherit from. `CreateGroupAPISchema` makes `name` required but `UpdateGroupAPISchema`. So in both `GroupAPISchema` and `UpdateGroupAPISchema` and empty dict is valid data. When adding manual code about `name` to `validate()` it would be easy to write code that crashes if there's no `name` in the given data, so adding this test to cover that case.